Extension/src/LanguageServer/extension.ts

```diff
@@ -199,6 +222,253 @@ function onInterval(): void {
+            await execSync(unInstallCommand);
+        }
+        await execSync(installCommand);
+    } catch (error) {
+        telemetry.logLanguageServerEvent("installCommandFailure");
+    }
+}
+
+async function downloadUrlForPlatform(build: Build): Promise<string | undefined> {
+    // Get the VSIX name to search for in build
+    const platformInfo: PlatformInformation = await PlatformInformation.GetPlatformInformation();
+    const vsixName: string | undefined = function(platformInfo): string {
+        switch (platformInfo.platform) {
+            case 'win32': return 'cpptools-win32.vsix';
+            case 'darwin': return 'cpptools-osx.vsix';
+            default: {
+                switch (platformInfo.architecture) {
+                    case 'x86': return 'cpptools-linux32.vsix';
+                    case 'x86_64': return 'cpptools-linux.vsix';
+                }
+            }
+        }
+    }(platformInfo);
+    if (!vsixName) {
+        return;
+    }
+
+    // Get the URL to download the VSIX, using vsixName as a key
+    const downloadUrl: string | undefined = build.assets.find((asset) => {
+        return asset.name === vsixName;
+    }).browserDownloadUrl;
+
+    return downloadUrl;
+}
+
+// Determines whether there exists a build that should be installed; returns the build if there is
+function getTargetBuild(releaseJson: Build[], updateChannel: string): Build | undefined {
+    // Get predicates to determine the build to install, if any
+    let needsUpdatePred: any;
+    let buildPred: any;
+    if (updateChannel === 'Insiders') {
+        needsUpdatePred = parsedVersionGreater;
+        buildPred = function(build: Build): boolean { return true; };
+    } else
+    if (updateChannel === 'Default') {
+        needsUpdatePred = function(v1: ParsedVersion, v2: ParsedVersion): boolean { return parsedVersionGreater(v2, v1); };
+        buildPred = function(build: Build): boolean { return build.name.indexOf('-') === -1; };
+    } else {
+        return;
+    }
+
+    // Get the build to install
+    const targetBuild: Build = releaseJson.find((build) => {
+        return buildPred(build);
+    });
+    if (!targetBuild) {
+        return;
+    }
+
+    // Check current version against target's version to determine if the installation should happen
+    const userVersion: ParsedVersion = getParsedVersion(util.packageJson.version);
+    const targetVersion: ParsedVersion = getParsedVersion(targetBuild.name);
+    if (!userVersion || !targetVersion) {
+        return;
+    }
+    if (!needsUpdatePred(userVersion, targetVersion)) {
+        return;
+    }
+
+    return targetBuild;
+}
+
+async function getReleaseJson(): Promise<Build[] | undefined> {
+    const releaseJsonFile: any = tmp.fileSync();
+
+    // Download json from GitHub
+    await util.downloadFileToDestination('https://api.github.com/repos/Microsoft/vscode-cpptools/releases',
+        releaseJsonFile.name, { 'User-Agent': 'vscode-cpptools' }).catch(() => {
```

Comment:
nit: make the url a const var so that you can fit all the parameters on one line.